### PR TITLE
Initial mobile swift tests

### LINKED ISSUE #83: Initial Mobile Swift Tests

### PR DESCRIPTION
**Pull request by martynkaqhe**
This PR introduces the initial unit testing infrastructure for the iOS (Swift) frontend to ensure proper integration with the existing Codecov CI pipeline. All GitHub Actions workflows remain untouched, relying on the pre-existing configuration.

### Details
**Configuration:** Updated mobile/Package.swift to include a new .testTarget named AppModuleTests linked to the main AppModule.

**Test Infrastructure:** Added the mobile/Tests/ directory with XCTest setup.

**Frontend Tests:** Created tests/GlaucomaViewModelTests.swift which includes comprehensive test plans for the GlaucomaViewModel:

**Risk Classifications:** Verified that riskText(cdr:) accurately returns the correct UI strings (NISKIE, UMIARKOWANE, WYSOKIE) based on CDR thresholds.

**Color Mapping:** Verified that riskColor(cdr:) correctly maps to .positiveGreen, .warningAmber, and .dangerRed.

**State Management:** Added tests to ensure the initial state is clean (test_InitialState_IsCorrect) and that the reset() function properly clears the session (test_Reset_ClearsState).

Coderabbit's summary:

- Added `.testTarget` named `AppModuleTests` to `mobile/Package.swift` that depends on `AppModule` and uses the `Tests` directory as its source path.
- Created `mobile/Tests/GlaucomaViewModelTests.swift` with a new `TestGlaucomaViewModel` test suite covering:
  - Risk text classification tests for CDR thresholds: "NISKIE" (< 0.4), "UMIARKOWANE" (0.4–0.5), "WYSOKIE" (≥ 0.6), including boundary cases.
  - Risk color mapping tests: `.positiveGreen` (< 0.4), `.warningAmber` (0.4–<0.6), `.dangerRed` (≥ 0.6), including boundary cases.
  - Initial state verification: `selectedImage` is nil, `analysisStep` is 0, `showPicker` is false.
  - Reset behavior verification: `reset()` clears state by resetting `selectedImage` to nil and `analysisStep` to 0.

[Review Change Stack](https://app.coderabbit.ai/change-stack/kequel/jaskra/pull/90)

### CODE REVIEW

File: mobile/Tests/GlaucomaViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        let result = viewModel.riskText(cdr: 0.5)
+        XCTAssertEqual(result, "UMIARKOWANE", "CDR between 0.4 and 0.6 should return moderate (UMIARKOWANE) classification")
+    }
+    
+    /// Test plan:
+    /// - Verify that riskText returns correct string for positive case (CDR >= 0.6)
+    func test_RiskText_PositiveCase_ReturnsHigh() {
+        let result = viewModel.riskText(cdr: 0.7)
+        XCTAssertEqual(result, "WYSOKIE", "CDR above 0.6 should return positive (WYSOKIE) classification")
+    }
```
[the remark this answers is not in the document]
Comment (martynkaqhe) [reply mobile/Tests/GlaucomaViewModelTests.swift]: Done

File: mobile/Tests/GlaucomaViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        let result = viewModel.riskColor(cdr: 0.5)
+        XCTAssertEqual(result, .warningAmber, "Moderate risk should return warningAmber color")
+    }
+    
+    /// Test plan:
+    /// - Verify riskColor returns dangerRed for positive case
+    func test_RiskColor_PositiveCase_ReturnsDangerRed() {
+        let result = viewModel.riskColor(cdr: 0.7)
+        XCTAssertEqual(result, .dangerRed, "High risk should return dangerRed color")
+    }
```
[the remark this answers is not in the document]
Comment (martynkaqhe) [reply mobile/Tests/GlaucomaViewModelTests.swift]: Done

[the remark this answers is not in the document]
Comment (martynkaqhe) [reply mobile/Tests/GlaucomaViewModelTests.swift]: Done

File: mobile/Tests/GlaucomaViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        viewModel.selectedImage = UIImage()
+        viewModel.analysisStep = 3
+        
+        // When
+        viewModel.reset()
+        
+        // Then
+        XCTAssertNil(viewModel.selectedImage, "Selected image should be nil after reset")
+        XCTAssertEqual(viewModel.analysisStep, 0, "Analysis step should be reset to 0")
+    }
```
[the remark this answers is not in the document]
Comment (martynkaqhe) [reply mobile/Tests/GlaucomaViewModelTests.swift]: Done

**Review by kequel — APPROVED** [review]
I like it much, but we will have to change README for tests

**Review by Agnesix — APPROVED** [review]
